src/tikzplotly/_dataContainer.py

```diff
@@ -61,28 +85,57 @@ def add_data(self, x, y, y_label=None):
             y values of the data
         y_label, optional
             name of the y data, by default None
+        name, optional
+            name of the data, by default None
 
         Returns
         -------
             tuple (macro_name, y_label), where macro_name is the name of the data in LaTeX and y_label the name of the y data in LaTeX
         """
         for data in self.data:
+            if isinstance(x, (tuple, list)):
+                continue
             if len(data.x) != len(x):
                 continue
             are_equals = data.x == x
             if isinstance(are_equals, bool):
                 if are_equals:
-                    y_label = data.add_y_data(y, y_label)
-                    return data.macro_name, y_label
-            elif are_equals.all():
-                if (data.x == x).all():
-                    y_label = data.add_y_data(y, y_label)
-                    return data.macro_name, y_label
-        data_to_add = Data(f"data{len(self.data)}", x)
-        y_label = data_to_add.add_y_data(y, y_label)
+                    y_label_val = data.add_y_data(y, y_label or name)
+                    return data.macro_name, treat_data(y_label_val)
+            elif hasattr(are_equals, "all") and are_equals.all():
+                y_label_val = data.add_y_data(y, y_label or name)
+                return data.macro_name, treat_data(y_label_val)
+        data_to_add = Data(f"data{index_to_letters(len(self.data))}", x)
+        y_label_val = data_to_add.add_y_data(y, y_label or name)
         self.data.append(data_to_add)
-        return data_to_add.macro_name, sanitize_text(y_label)
+        return data_to_add.macro_name, treat_data(y_label_val)
```

Comment:
We should be caution about it : there are two type of texts : - TeX texts, used for legend, title etc
- other texts, for data labels, etc
The handling of these texts is different, for instance in `test_sanitized_text`, a ployly label with `[test​]` as label shouldn't be included as `[test]` in column label, as it won't compile with LaTeX

Reply:
fixed in d9b96ec941f45f886d897bfba9222be6fcc31b10